src/python/pants/engine/target.py

```diff
@@ -227,6 +227,9 @@ def __init__(self, raw_value: Optional[Any], *, address: Address) -> None:
         # We must temporarily unfreeze the field, but then we refreeze to continue avoiding
         # subclasses from adding arbitrary fields.
         self._unfreeze_instance()  # type: ignore[attr-defined]
+        # N.B.: We store the address here and not in the Field base class, because the memory usage
+        # of storing this value in every field was shown to be excessive / lead to performance
```

Comment:
Let me know if either half of the `/` is a lie and I can reword.

Reply:
I don't think we benchmarked it, more based on reasoning. Back-of-the-envelope math:

- the target types we have seem to roughly be 3-8 fields, so let's average to 5 fields/tgt.
- imagine the user operates on 500 targets
- then we would be storing 2500 extra `Address` objects, in addition to the 500 already stored on the owning `Target` objects.

An insight from the original AsyncField concept: when we use fields in rules, we want to be able to operate on that single Field instance without needing to consider the rest of the target's Fields. This improves cache keys. So those fields must have an Address instance; but when we're not using the Rules API, we can simply say `tgt[MyField]` and we can easily get the address from the original target.